tests/admin_scripts/tests.py

```diff
@@ -2116,19 +2116,26 @@ def test_invalid_name(self):
 
     def test_invalid_target_name(self):
         """startapp validates that target dir name is a valid Python identifier."""
-        os.mkdir('app_dir')
-        self.addCleanup(shutil.rmtree, 'app_dir')
-
-        for bad_target in ('app_dir/invalid.name', 'app_dir/7invalid_name'):
+        for bad_target in ('invalid.dir_name', '7invalid_dir_name'):
             args = ['startapp', 'valid_name', bad_target]
+
             os.mkdir(bad_target)
 
             out, err = self.run_django_admin(args)
+            shutil.rmtree(bad_target)
             self.assertOutput(
                 err,
-                "CommandError: '{}' is not a valid app name. Please make sure the "
-                "name is a valid identifier.".format(os.path.basename(bad_target))
+                "CommandError: '{}' is not a valid app directory. Please make sure the "
+                "directory is a valid identifier.".format(bad_target)
             )
+        args = ['startapp', 'valid_name', 'os']
```

Comment:
I would move test for conflict with an existing Python module to a separate test e.g. `test_importable_target_name`